pulsar-broker/src/main/java/com/yahoo/pulsar/broker/service/BrokerService.java

```diff
@@ -824,4 +841,76 @@ public AuthenticationService getAuthenticationService() {
     public List<PersistentTopic> getAllTopicsFromNamespaceBundle(String namespace, String bundle) {
         return multiLayerTopicsMap.get(namespace).get(bundle).values();
     }
+    
+    public ZooKeeperDataCache<Map<String, String>> getDynamicConfigurationCache() {
+        return dynamicConfigurationCache;
+    }
+
+    /**
+     * Update dynamic-ServiceConfiguration with value present into zk-configuraiton-map and register listeners on
+     * dynamic-ServiceConfiguration field to take appropriate action on change of zk-configuraiton-map.
+     */
+    private void updateConfigurationAndRegisterListeners() {
+        // update ServiceConfiguration value by reading zk-configuration-map
+        updateDynamicServiceConfiguration();
+        // update brokerShutdownTimeoutMs value on listener notification
+        registerConfigurationListener("brokerShutdownTimeoutMs", new Consumer<String>() {
+            @Override
+            public void accept(String shutdownTime) {
```

Comment:
>Actually, for this example, we shouldn't even need to register for notification, right?

Actually, we are not updating `ServiceConfiguration` instance's field value but we are calling registered listener for this field and it's registered-listener's responsibility to update the value and take appropriate action. 
Because in case of #181 : Only when `lookupRequestPermits` field-value changes we want to take certain action such as update `Semaphore`'s permits. Now, we have watch on entire [dynamicConfigMap](https://github.com/yahoo/pulsar/pull/186/files/206ba5d2d44275a6e061fffa1a791ab7580b0c8e#diff-483f002e2db69883012f75649177d1edR142) and listener can be triggered on every configKey-value change. So, registered listener can compare existing value and new value, and if it's changed then it can take appropriate action and update `ServiceConfiguration`.